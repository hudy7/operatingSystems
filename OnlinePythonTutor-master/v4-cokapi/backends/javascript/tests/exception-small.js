try {
    throw Error('oops');
}
catch (e) {
    var xxx = 1;
    console.log(e); // e should be in scope here
}
