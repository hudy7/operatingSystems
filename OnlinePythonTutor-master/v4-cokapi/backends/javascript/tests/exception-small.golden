{ code: 'try {\n    throw Error(\'oops\');\n}\ncatch (e) {\n    var xxx = 1;\n    console.log(e); // e should be in scope here\n}',
  trace: 
   [ { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { xxx: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'xxx' ],
       line: 2,
       col: 4,
       event: 'step_line',
       heap: {} },
     { exception_msg: 'Error: oops',
       stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { xxx: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'xxx' ],
       line: 2,
       col: 4,
       event: 'exception',
       heap: {} },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { e: [ 'REF', 1 ], xxx: [ 'JS_SPECIAL_VAL', 'undefined' ] },
       ordered_globals: [ 'e', 'xxx' ],
       line: 5,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'INSTANCE_PPRINT', 'object', 'Error: oops' ] } },
     { stdout: '',
       func_name: '',
       stack_to_render: [],
       globals: { e: [ 'REF', 1 ], xxx: 1 },
       ordered_globals: [ 'e', 'xxx' ],
       line: 6,
       col: 4,
       event: 'step_line',
       heap: { '1': [ 'INSTANCE_PPRINT', 'object', 'Error: oops' ] } },
     { stdout: 'Error: oops\n    at Error (native)\n    at userscript.js:4:11\n    at ContextifyScript.Script.runInNewContext (vm.js:18:15)\n    at module.exports (/home/pgbovine/OnlinePythonTutor/v4-cokapi/backends/javascript/node_modules/eval/eval.js:56:8)\n    at Object.<anonymous> (/home/pgbovine/OnlinePythonTutor/v4-cokapi/backends/javascript/jslogger.js:986:3)\n    at Module._compile (module.js:541:32)\n    at Object.Module._extensions..js (module.js:550:10)\n    at Module.load (module.js:456:32)\n    at tryModuleLoad (module.js:415:12)\n    at Function.Module._load (module.js:407:3)\n',
       func_name: '',
       stack_to_render: [],
       globals: { xxx: 1 },
       ordered_globals: [ 'xxx' ],
       line: 7,
       col: 1,
       event: 'step_line',
       heap: {} } ] }
